### DISCUSSION TITLE
White screen after packaging windows app

### DISCUSSION THREAD

**Discussion by MrDIDJoy** [flet-dev/flet#4559] [Q&A] [answered]
### Question

After packaging, the application does not load content, but simply displays a white screen, while in pycharm it works fine

https://github.com/user-attachments/assets/4c158527-a3e4-4996-a6e8-ab14e8328642

Logs: 
```
[13:55:15] Created Flutter bootstrap project from           
           gh:flet-dev/flet-build-template with ref 0.25.1  
           ✅                                               
           Customized app icons and splash images ✅        
[13:55:33] Generated app icons ✅                           
[13:55:37] Packaged Python app ✅                           
[13:59:44] Built Windows app ✅                             
[13:59:45] Copied build to build\windows directory ✅       
           Successfully built your Windows app! 🥳 Find it  
           in build\windows directory. 📁 
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by MrDIDJoy** [comment]
im use >>> flet build windows

**Reply by ndonkoHenri** [reply]
what happens when you package a helloworld app with Flet version 0.25.2 ?

**Reply by MrDIDJoy** [reply]
Thanks for the reply. Unfortunately, it's the same thing, but I noticed something. At the first launch after packaging _(I did not record the video)_, the program name was written on the appbar, and the screen was a little pink. After 2-3 seconds, the screen turned white, and a void appeared on the appbar instead of the program name. I also noticed that the file is suspiciously light in weight: only 71-72 kilobytes

https://github.com/user-attachments/assets/948568f8-a70b-4de3-b4e2-a1c0790f0486

![Скриншот 14-12-2024 183727](https://github.com/user-attachments/assets/e8f67789-678b-40e4-8ffc-55f823f4a6dc)

![Скриншот 14-12-2024 184411](https://github.com/user-attachments/assets/7a2cc67b-2ea0-4618-90ab-3c02565f8080)

**Reply by MrDIDJoy** [reply]
I also noticed that there is not a single flet folder in the build\windows\Lib folder, perhaps this is because of this. It may even be due to incomplete installation of Visual Studio packages. Please remind me which ones are needed for correct operation? Maybe I missed something.

requirements.txt from project:
```
altgraph==0.17.4
annotated-types==0.7.0
anyio==4.6.2.post1
arrow==1.3.0
binaryornot==0.4.4
certifi==2024.8.30
cffi==1.17.1
chardet==5.2.0
charset-normalizer==3.4.0
click==8.1.7
colorama==0.4.6
cookiecutter==2.6.0
cryptography==43.0.3
fastapi==0.115.5
fernet==1.0.1
flet==0.25.1
flet-cli==0.25.1
flet-core==0.24.1
flet-desktop==0.25.1
flet-runtime==0.24.1
flet-web==0.25.1
h11==0.14.0
httpcore==1.0.6
httptools==0.6.4
httpx==0.27.2
idna==3.10
Jinja2==3.1.4
lock==2018.3.25.2110
markdown-it-py==3.0.0
MarkupSafe==3.0.2
mdurl==0.1.2
Nuitka==2.5.6
oauthlib==3.2.2
ordered-set==4.1.0
packaging==23.2
pefile==2023.2.7
pillow==11.0.0
pyaes==1.6.1
pycparser==2.22
pydantic==2.9.2
pydantic_core==2.23.4
Pygments==2.18.0
pyinstaller==6.11.1
pyinstaller-hooks-contrib==2024.10
pypng==0.20220715.0
PyQt5==5.15.11
PyQt5-Qt5==5.15.2
PyQt5_sip==12.16.1
python-dateutil==2.9.0.post0
python-dotenv==1.0.1
python-slugify==8.0.4
pywin32-ctypes==0.2.3
PyYAML==6.0.2
qrcode==7.4.2
repath==0.9.0
requests==2.32.3
rich==13.9.4
setuptools==75.6.0
six==1.16.0
sniffio==1.3.1
starlette==0.41.2
text-unidecode==1.3
toml==0.10.2
types-python-dateutil==2.9.0.20241003
typing_extensions==4.12.2
urllib3==2.2.3
uvicorn==0.32.0
validators==0.34.0
watchdog==4.0.2
watchfiles==0.24.0
websockets==14.0
zstandard==0.23.0
```

**Reply by ndonkoHenri** [reply]
Your requirements file shouldnt be made with pip freeze, as mentioned in the [docs](https://flet.dev/docs/publish/#project-structure:~:text=as%20watchdog.-,Hand,-%2Dpick%20requirements.txt).
You need to have only the packages used in your app. Ex:
```bash
flet
pillow==X.Y
...
...
```

Also, as I said, I will like you to try a simple hello world app to see if it works before trying with your project.

<details><summary>Details</summary>
<p>

`main.py`

```py
import flet as ft

def main(page: ft.Page):
    page.add(ft.Text("Hello World!"))

ft.app(main)
``` 

And in `requirements.txt` just "flet"

</p>
</details>

**Reply by MrDIDJoy** [reply]
Everything is working fine with the test program. Most likely, in my main program, this happened due to the fact that in requirements.txt I had too many libraries. I will try to keep only the ones that are in use. By the way, could you please give me an example of pyproject.toml? On the internet, I couldn't find it.

**Reply by ndonkoHenri** [reply]
You can get inspiration from [here](https://flet.dev/docs/publish#versioning).

**Reply by MrDIDJoy** [reply]
requirements update don't helped :/

**Reply by MrDIDJoy** [reply, marked as answer]
I did it. The problem was that the assets folder was not in the build\windows folder. After I copied it along with the icon from the project, everything started fine! Even though I figured out what the problem was, thank you so much for trying to help me. I like your framework and wish it further successful development!

**Comment by mrbattery1130** [comment]
Hello, I've encountered the same issue.
Could you please tell me the specific location within the `build\windows` directory which you copied the contents of the `assets` folder to?Or could you show me the directory tree as an example?
Thank you.
